app/parse.py

```
@@ -15,8 +33,108 @@ class Product:
     num_of_reviews: int
 
 
+def set_driver() -> Options:
+    options = Options()
+    options.add_argument("--headless")
+    options.add_argument("--log-level=3")
+    options.add_argument("--silent")
+    return options
+
+
+def accept_cookies(banner: WebElement) -> None:
+    accept_button = banner.find_element(
+        By.CSS_SELECTOR, ".acceptContainer > button.acceptCookies"
+    )
+    accept_button.click()
+
+
+def get_products(element: WebElement, category: str) -> Product:
+    return Product(
+        title=element.find_element(
+            By.CSS_SELECTOR, ".caption > h4 > a"
+        ).get_attribute("title"),
+        description=element.find_element(
+            By.CSS_SELECTOR, ".caption > p.description"
+        ).text,
+        price=float(element.find_element(
+            By.CSS_SELECTOR, ".caption > h4.float-end"
+        ).text.replace("$", "")),
+        rating=len(element.find_elements(
+            By.CSS_SELECTOR, ".ratings span.ws-icon-star")
+        ),
+        num_of_reviews=int(
+            element.find_element(
+                By.CSS_SELECTOR, ".ratings > p.review-count"
+            ).text.split(" ")[0]),
+    )
+
+
+def save_in_csv(products: [Product], filename: str) -> None:
+    with open(f"{filename}.csv", "w", encoding="utf-8", newline="") as f:
+        writer = csv.writer(f)
+        writer.writerow(["title", "description", "price", "rating", "num_of_reviews"])
+        for product in products:
+            writer.writerow([
+                product.title,
+                product.description,
+                product.price,
+                product.rating,
+                product.num_of_reviews,
+            ])
+
+
+def parse_product_page(driver: WebDriver, url: str, category: str) -> [Product]:
+    driver.get(urljoin(BASE_URL, url))
+    products = []
+    try:
+        cookie_banner = driver.find_element(By.ID, "cookieBanner")
+        accept_cookies(cookie_banner)
+    except:
```

avoid the bare except + pass antipattern; try to be specific in except